freqtrade/freqai/data_kitchen.py

```diff
@@ -1344,3 +1347,24 @@ def check_if_backtest_prediction_is_valid(
                 f"Could not find backtesting prediction file at {path_to_predictionfile}"
             )
             return False
+
+    def reduce_dataframe_footprint(self, df: DataFrame) -> DataFrame:
+        """
+        Ensure all values are float32
+        """
+        start_mem = df.memory_usage().sum() / 1024**2
+        print("Memory usage of dataframe is {:.2f} MB".format(start_mem))
+
+        for col in df.columns[1:]:
+            col_type = df[col].dtype
+            if col_type != object:
+                if str(col_type)[:3] == "int":
+                    df[col] = df[col].astype(np.int32)
+                else:
+                    df[col] = df[col].astype(np.float32)
```

Comment:
i think you'd want an `elif` here with `isinstance(df[col], np.floating)` (i'd also do a comparison against `np.integer` above in a similar fashion).

you can't outright assume that everything that's not an int is a float - it could just as well be a bool (explicitly assigned type) - or a string (`dataframe['dir'] = 'raising'`)

Reply:
looks resolved.